_posts/2022-12-12-trino-summit-2022-lyft-recap.md

```diff
@@ -0,0 +1,57 @@
+---
+layout: post
+title: "Trino for large scale ETL at Lyft"
+author: "Charles Song, Ritesh Varyani, Brian Olsen"
+excerpt_separator: <!--more-->
+image: /assets/blog/trino-summit-2022/lyft.jpg
+---
+Buckle up, for the next [post in the Trino Summit 2022 recap series]({% post_url
+2022-11-21-trino-summit-2022-recap %}). In this post, we're covering the talk
+given by Lyft engineers, Charles and Ritesh, on how they have not only scaled
+Trino for more usage, but also with less nodes and more effective usage. They
+also started moving to utilizing Trino more for ETL rather than just interactive
+analytics. You don't want to miss your driver for this post.
+
+<!--more-->
+
+{% youtube FL3c1Ue7YWM %}
+
+<a class="btn btn-pink btn-md" target="_blank" href="/assets/blog/trino-summit-2022/Trino@Lyft.pdf">
+  Check out the slides here!
+</a>
+
+## Recapping
+
+Lyft is uses Trino to perform ETL jobs reading 10 petabytes of data per day and
+writing 100 terabytes per day. They run 250,000 queries per day, with around
+2,000 unique users. This requires approximately 750 EC2 instances scaling up or
+down with an auto-scalar. Over 90 percent of queries run within a one to three
+minutes. 
+
+In the last year, Lyft cut their number of Trino nodes in half, while increasing
+the volume of their workloads due to recent improvements in Trino and upgrades
+in Java versions. Lyft does not using fault-tolerant execution, but has started
+seeing interest in using Trino for ETL jobs due to the faster turnaround. Some
+issues Lyft has faced has been around how resource hungry Trino is, as well as,
+the issue where the coordinator can be a single point of failure for queries
+executing on a cluster.
+
+Lyft was one of the earliest companies to really push using Trino for ETL use
+cases. They built custom best effort rollback code in Airflow. If a query
+failed, the operation would revert to the state before the operation began. Lyft
+runs four ETL Trino clusters split by the type of workload used on that cluster.
```

Comment:
so they are all ETL .. but still different workload?

Reply:
Nope, I removed ETL ahead of Trino